lib/charms/hydra/v0/oauth.py

```diff
@@ -335,13 +354,23 @@ def __init__(
         events = self._charm.on[relation_name]
         self.framework.observe(events.relation_created, self._on_relation_created_event)
         self.framework.observe(events.relation_changed, self._on_relation_changed_event)
+        self.framework.observe(events.relation_broken, self._on_relation_broken_event)
 
     def _on_relation_created_event(self, event: RelationCreatedEvent) -> None:
         try:
             self._update_relation_data(self._client_config, event.relation.id)
         except ClientConfigError as e:
             self.on.invalid_client_config.emit(e.args[0])
 
+    def _on_relation_broken_event(self, event: RelationBrokenEvent) -> None:
+        if self.is_client_created():
+            event.defer()
+            logger.info("Relation data still available. Deferring the event")
+            return
+
```

Comment:
Why don't we pop the relation data like in the provider side?

Reply:
I focused on removing the provider's data, but that's a good point, I'll pop the requirer too.